### ISSUE TITLE
Getting an error when trying to run some examples

### ISSUE THREAD

**Issue by stefanstranger** [dfinke/edge-powershell#7]
When I run the following from the samples folder I get the following error:

D:\Install\Node\edge-powershell-master\samples\tryPowerShell> node .\test.js

D:\Install\Node\edge-powershell-master\samples\tryPowerShell\test.js:16
    if (error) throw error;
                     ^
System.AggregateException: One or more errors occurred. ---> System.InvalidCastException: Unable to cast object of type
'System.String' to type 'System.Collections.Generic.IDictionary`2[System.String,System.Object]'.
   at Edge.PS.EdgePowerShell.<InvokeScript>d__2.MoveNext() in d:\Install\Node\edge-powershell-master\src\Edge.PowerShell
\Edge.PowerShell\EdgePowerShell.cs:line 36
   --- End of inner exception stack trace ---
---> (Inner Exception #0) System.InvalidCastException: Unable to cast object of type 'System.String' to type 'System.Col
lections.Generic.IDictionary`2[System.String,System.Object]'.
   at Edge.PS.EdgePowerShell.<InvokeScript>d__2.MoveNext() in d:\Install\Node\edge-powershell-master\src\Edge.PowerShell
\Edge.PowerShell\EdgePowerShell.cs:line 36<---

What am I doing wrong?

I've installed Node.js 0.8.19 and Edge.

/Stefan

**Comment by dfinke** [comment]
Thanks for trying Stefan. I'll get back to you shortly after I investigate.

**Comment by stefanstranger** [comment]
Thnx

-----Original Message-----

**Comment by dfinke** [comment]
Ok, sorry about that. I was in the middle of spiking out some ideas and checked them in. This version spins up a web server and serves up PowerShell files. So... Try this

```
node .\server.js
```

Then launch a browser and use this url

```
http://localhost:8080/powershell/test
```

You should get this

```
[
  4,
  16,
  36,
  64,
  100,
  144,
  196,
  256,
  324,
  400
]
```

You can see that it is executing test.ps1.  It can also take two parameters like this.

```
http://localhost:8080/powershell/test?max=100&min=90
```

```
[
  8100,
  8464,
  8836,
  9216,
  9604,
  10000
]
```

You can create your own PowerShell files. Save them in the _edge-powershell\samples\tryPowerShell_ directory. You execute it with the base url 

```
http://localhost:8080/powershell
```

Then the base name of the file without the ps1 extension.

```
http://localhost:8080/powershell/myscript
```

Passing parameters follow the query string convention

```
http://localhost:8080/powershell/myscript?firstname=john&lastname=doe
```

I'm working towards merging the solution where you can run this as standalone or through a web server.

**Comment by cdhunt** [comment]
Installed Express and set $env:PORT to an available port and it worked as expected. 

So, I created this script.

```
param($script = 'Get-Process')

Invoke-Expression $script
```

It throws an error

```
                res.json(results.Result);
                                ^
TypeError: Cannot read property 'Result' of undefined
```

Simple scripts like "1..3" work. Maybe this has something to do with #6.

If I pipe to `Select-Object -Property *` it doesn't crash but returns an array of empty json objects.

```
[
  {},
  {},
  {},
  {},
  {}
]
```

**Comment by dfinke** [comment]
I believe it may work if you pipe it to Out-String

**Comment by cdhunt** [comment]
Yes, that works as well as ConvertTo-JSON, but like you said, that's just a big string, not an object.

**Comment by dfinke** [comment]
Yep, so getting an "object" back is the challenge. I'm not sure how, I have some guesses. I'm wondering if/how PowerShell Web Access does it.
